napari/layers/shapes/shapes.py

```diff
@@ -2617,12 +2628,24 @@ def _transform_box(self, transform, center=(0, 0)):
         box = self._selected_box - center
         box = box @ transform.T
         if not np.all(box[Box.TOP_CENTER] == box[Box.HANDLE]):
-            r = self._rotation_handle_length * self.scale_factor
+            r = (
+                self._rotation_handle_length
+                * self.scale_factor
+                / self.scale[-1]
+            )
             handle_vec = box[Box.HANDLE] - box[Box.TOP_CENTER]
             cur_len = np.linalg.norm(handle_vec)
             box[Box.HANDLE] = box[Box.TOP_CENTER] + r * handle_vec / cur_len
         self._selected_box = box + center
 
+    def _update_draw(
```

Comment:
@brisvag I think I get everything, but I don't get this—it doesn't seem used?
Can you clarify for my edification?

Reply:
This is overloading the base layer. It's called every time the zoom level changes, which updates `self.scale_factor`. We need this change to propagate to the highlight (depending on zoom we need to decrease the size of the mesh that makes up the highlight)